File: src/main/java/io/gravitee/repository/model/Application.java
/**
 * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.gravitee.repository.model;

import java.util.Date;
import java.util.Objects;

/**
 * @author David BRASSELY (brasseld at gmail.com)
 */
public class Application {

	/**
	 * The application name
	 */
    private String name;
    
    /**
     * The application description
     */
    private String description;
    
    private String type;

    /**
     * The team creation date
     */
    private Date createdAt;
    
    /**
     * The team last updated date
     */
    private Date updatedAt;

    /**
     * The owner entity (user or team)
     */
    private OwnerType ownerType;
    
    /**
     * The owner entity name (user name or team name)
     */
    private String owner;
    
    /**
     * The user name that create the application
     */
    private String creator;
    
    /**
     * The application key
     */
    private String key;

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Date updatedAt) {
        this.updatedAt = updatedAt;
    }
    
    public OwnerType getOwnerType() {
		return ownerType;
	}

	public void setOwnerType(OwnerType ownerType) {
		this.ownerType = ownerType;
	}

	public String getOwner() {
		return owner;
	}

	public void setOwner(String owner) {
		this.owner = owner;
	}

    public String getCreator() {
        return creator;
    }

    public void setCreator(String creator) {
        this.creator = creator;
    }

    public String getKey() {
		return key;
	}

	public void setKey(String key) {
		this.key = key;
	}

	@Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Application that = (Application) o;
        return Objects.equals(name, that.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name);
    }

    @Override
    public String toString() {
        final StringBuilder sb = new StringBuilder("Application{");
        sb.append("name='").append(name).append('\'');
        sb.append(", type='").append(type).append('\'');
        sb.append('}');
        return sb.toString();
    }
}

File: gravitee-gamma/gravitee-gamma-module-apim/src/main/ui/portal-editor/editor/entities/application.ts
/*
 * Copyright (C) 2026 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import type { ApplicationOwner } from './member';

export type ApplicationType = 'SIMPLE' | 'SPA' | 'WEB' | 'NATIVE' | 'BACKEND_TO_BACKEND';

export interface GrantTypeConfig {
    type: string;
    name: string;
    response_types: string[];
}

export interface ApplicationTypeConfig {
    id: string;
    name: string;
    description: string;
    applicationType: ApplicationType;
    requires_redirect_uris: boolean;
    allowed_grant_types: GrantTypeConfig[];
    default_grant_types: GrantTypeConfig[];
    mandatory_grant_types: GrantTypeConfig[];
}

export interface ApplicationSettingsOAuth {
    client_id?: string;
    client_secret?: string;
    redirect_uris?: string[];
    grant_types?: string[];
    application_type?: string;
    additional_client_metadata?: Record<string, string>;
}

export interface ApplicationSettingsApp {
    client_id?: string;
    type?: string;
}

export interface ApplicationSettings {
    oauth?: ApplicationSettingsOAuth;
    app?: ApplicationSettingsApp;
}

export interface Application {
    id: string;
    name: string;
    description?: string;
    domain?: string;
    applicationType?: ApplicationType;
    api_key_mode?: 'EXCLUSIVE' | 'SHARED' | 'UNSPECIFIED';
    created_at?: string;
    updated_at?: string;
    owner?: ApplicationOwner;
    portalTenantId?: string;
    settings: ApplicationSettings;
}

export interface ApplicationsResponse {
    data: Application[];
    metadata?: {
        pagination?: {
            current_page: number;
            size: number;
            total: number;
            total_pages: number;
        };
    };
}
